### PR TITLE
[Network January] Disconnect VPN connections

### PR DESCRIPTION
**Pull request by ninweninwe**
## Description

Disconnect vpn connection for Virtual network gateway and P2S Vpn gateway.

Azure/azure-powershell-cmdlet-review-pr#414

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Network/Network.Test/ScenarioTests/CortexTests.ps1 [hunk 1]
```diff
@@ -683,6 +683,89 @@ function Test-CortexExpressRouteCRUD
...
+		Assert-AreEqual $P2SvpnGatewayName $P2SVpnGateway.Name
+		Assert-AreEqual "Succeeded" $P2SVpnGateway.ProvisioningState
+
+		$expected = Disconnect-AzP2SVpnGatewayVpnConnection -ResourceGroupName $rgname -ResourceName $P2SvpnGatewayName -VpnConnectionId @("IKEv2_1e1cfe59-5c7c-4315-a876-b11fbfdfeed4")
+        Assert-AreEqual $expected.Name $P2SVpnGateway.Name
+     }
+     finally
+     {
+		# Delete P2SVpnGateway using Remove-AzP2sVpnGateway
+		$delete = Remove-AzP2sVpnGateway -Name $P2SVpnGatewayName -ResourceGroupName $rgName -Force -PassThru
```
Comment (number213) [inline src/Network/Network.Test/ScenarioTests/CortexTests.ps1]: Please move those removes under `try`

File: src/Network/Network/ChangeLog.md [hunk 2]
```diff
@@ -37,6 +37,10 @@
         - New-AzApplicationGatewayRewriteRuleUrlConfiguration
     - Cmdlets updated with optional parameter - UrlConfiguration
         - New-AzApplicationGatewayRewriteRuleActionSet
+* Added support to disconnect vpn connection in virtual network gateway and p2s vpn gateway
```
Comment (number213) [inline src/Network/Network/ChangeLog.md]: Please move this under `Upcoming Release`

File: src/Network/Network/Cortex/P2SVpnGateway/DisconnectAzureP2SVpnGatewayVpnConnection.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByP2SVpnGatewayName, 
+            HelpMessage = "The resource name.")]
+        [ResourceNameCompleter("Microsoft.Network/p2sVpnGateways", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+        
+        [Parameter(
+            ParameterSetName = CortexParameterSetNames.ByP2SVpnGatewayResourceId,
+            Mandatory = false,
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/P2SVpnGateway/DisconnectAzureP2SVpnGatewayVpnConnection.cs]: this is true in the help file, why is false here

File: src/Network/Network/Cortex/P2SVpnGateway/DisconnectAzureP2SVpnGatewayVpnConnection.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The Azure resource ID of the P2SVpnGateway to be modified.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Alias("P2SVpnGateway")]
+        [Parameter(
+            ParameterSetName = CortexParameterSetNames.ByP2SVpnGatewayObject,
+            Mandatory = false,
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/P2SVpnGateway/DisconnectAzureP2SVpnGatewayVpnConnection.cs]: this is true in the help file, why is false here

File: src/Network/Network/Cortex/P2SVpnGateway/DisconnectAzureP2SVpnGatewayVpnConnection.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    using Microsoft.Azure.Management.Network.Models;
+    using System;
+    using System.Collections.Generic;
+    using System.Management.Automation;
+
+    [Cmdlet("Disconnect",
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "P2SVpnGatewayVpnConnection",
+        DefaultParameterSetName = CortexParameterSetNames.ByP2SVpnGatewayName, 
+        SupportsShouldProcess = true),
+        OutputType(typeof(PSP2SVpnGateway))]
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/P2SVpnGateway/DisconnectAzureP2SVpnGatewayVpnConnection.cs]: this doesn't seems to match the help file

File: src/Network/Network/VirtualNetworkGateway/DisconnectAzureVirtualNetworkGatewayVpnConnection.cs [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Network.VirtualNetworkGateway
+{
+    [Cmdlet("Disconnect", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualNetworkGatewayVpnConnection", DefaultParameterSetName = "ByFactoryName", SupportsShouldProcess = true), OutputType(typeof(PSVirtualNetworkGateway))]
+    public class DisconnectVirtualNetworkGatewayVpnConnectionCommand : VirtualNetworkGatewayBaseCmdlet
```
Comment (VeryEarly) [inline src/Network/Network/VirtualNetworkGateway/DisconnectAzureVirtualNetworkGatewayVpnConnection.cs]: This doesn't seem to match the help file

File: src/Network/Network/VirtualNetworkGateway/DisconnectAzureVirtualNetworkGatewayVpnConnection.cs [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ParameterSetNames.ByFactoryName,
+            HelpMessage = "Virtual network gateway name")]
+        [ResourceNameCompleter("Microsoft.Network/virtualNetworkGateways", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public virtual string VirtualNetworkGatewayName { get; set; }
+
+        [Alias("VirtualNetworkGatewayId")]
+        [Parameter(
+            Mandatory = false,
```
Comment (VeryEarly) [inline src/Network/Network/VirtualNetworkGateway/DisconnectAzureVirtualNetworkGatewayVpnConnection.cs]: this is true in the help file, why is false here

File: src/Network/Network/VirtualNetworkGateway/DisconnectAzureVirtualNetworkGatewayVpnConnection.cs [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ParameterSetNames.ByResourceId,
+            HelpMessage = "The resource id of the virtual network gateway Id")]
+        public string ResourceId { get; set; }
+
+        [Alias("VirtualNetworkGateway")]
+        [Parameter(
+            Mandatory = false,
```
Comment (VeryEarly) [inline src/Network/Network/VirtualNetworkGateway/DisconnectAzureVirtualNetworkGatewayVpnConnection.cs]: this is true in the help file, why is false here

**Comment by VeryEarly** [comment]
[Parameter(Mandatory = false)]
public SwitchParameter PassThru { get; set; }

are missing for both cmdlets

**Pull request merged**